### PR TITLE
usm: Fix event consumer registration

### PR DESCRIPTION
**Pull request by vitkyrka**
### What does this PR do?

Currently, the process monitor event stream consumer is registered
if EnableUSMEventStream is set, but there are other options which
determine if the process monitor really is used.  If the process monitor
is not used and the event stream consumer is registered, there is a
segfault when events are received since the process monitor has not been
initialized:

```
 panic: runtime error: invalid memory address or nil pointer dereference
 [signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x2eaba1c]
 goroutine 2168 [running]:
 pkg/network/protocols/telemetry.(*Counter).Add(...)
 pkg/process/monitor.(*EventConsumer).HandleEvent()
 pkg/security/probe.(*EventConsumer).Start.func1()
 created by pkg/security/probe.(*EventConsumer).Start in goroutine 1
```

To fix this, ensure that the conditions for registering the event stream
consumer and initializing the process monitor are the same.

### Motivation

[Segfault in staging](https://ddstaging.datadoghq.com/logs?query=service%3Asystem-probe%20status%3Aerror%20panic%20-nodegroup%3Adist-aggr_dist-aggr-c4&agg_m=count&agg_m_source=base&agg_t=count&cols=host%2Cservice&event=AgAAAZB4309hd6slhAAAAAAAAAAYAAAAAEFaQjQzMVlkQUFEV0hMVk5ScGhVYmdBZwAAACQAAAAAMDE5MDc4ZTAtYTY3OC00ZmRmLWFiNjAtNzMzMjZmZGYyNjhk&fromUser=true&messageDisplay=inline&refresh_mode=paused&storage=hot&stream_sort=desc&viz=stream&from_ts=1720014600000&to_ts=1720015200000&live=false)
https://datadoghq.atlassian.net/browse/USMON-1069

### CODE REVIEW

**Review by paulcacheux — APPROVED** [review]

File: pkg/network/usm/config/config.go [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+// Package config provides helpers for USM configuration
+package config
+
+import (
+	"github.com/DataDog/datadog-agent/pkg/network/config"
+	"github.com/DataDog/datadog-agent/pkg/network/protocols/http"
+)
+
+// NeedUSM returns true if USM is supported and enabled
+func NeedUSM(config *config.Config) bool {
```
Comment (DanielLavie) [inline pkg/network/usm/config/config.go:17]: nit: consider renaming the function to better convey its purpose of checking support and enabling status (e.g., isUSMSupportedAndEnabled)

**Review by DanielLavie — APPROVED** [review]

File: pkg/network/usm/config/config.go (hunk 1 above)
Comment (vitkyrka) [reply pkg/network/usm/config/config.go]: Done.

**Comment by paulcacheux** [comment]
The `Go Package Import Differences` shows some issue I believe

File: cmd/system-probe/modules/eventmonitor_linux.go [hunk 2]
```diff
@@ -25,6 +27,10 @@ var EventMonitor = module.Factory{
...
 }
 
-func createProcessMonitorConsumer(evm *eventmonitor.EventMonitor) (eventmonitor.EventConsumerInterface, error) {
+func createProcessMonitorConsumer(evm *eventmonitor.EventMonitor, config *netconfig.Config) (eventmonitor.EventConsumerInterface, error) {
+	if !usmconfig.IsUSMSupportedAndEnabled(config) || !usmconfig.NeedProcessMonitor(config) {
+		return nil, nil
+	}
+
 	return procmon.NewProcessMonitorEventConsumer(evm)
 }
```
Comment (guyarb) [inline cmd/system-probe/modules/eventmonitor_linux.go:36]: Is it only for USM?

**Review by guyarb — APPROVED** [review]

**Review by hmahmood — APPROVED** [review]

File: cmd/system-probe/modules/eventmonitor_linux.go (hunk 2 above)
Comment (vitkyrka) [reply cmd/system-probe/modules/eventmonitor_linux.go:36]: Yes, this particular consumer is only used for USM at the moment.

**Pull request merged**